apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/spi/defaultimpl/DefaultRoleInitializationService.java

```diff
@@ -34,38 +35,58 @@
   @Autowired
   private PortalConfig portalConfig;
 
+  /**
+   * 初始化应用角色
+   * @param app
+   */
+  @Override
   @Transactional
   public void initAppRoles(App app) {
-    String appId = app.getAppId();
 
+    String appId = app.getAppId();
+    // 创建 App 拥有者的角色名
     String appMasterRoleName = RoleUtils.buildAppMasterRoleName(appId);
 
-    //has created before
+    // 校验角色是否已经存在。若是，直接返回
     if (rolePermissionService.findRoleByRoleName(appMasterRoleName) != null) {
       return;
     }
     String operator = app.getDataChangeCreatedBy();
+
     //create app permissions
     createAppMasterRole(appId, operator);
 
+    // 初始化查看角色
+    createAppViewRole(appId, operator);
+    List<Env> portalEnvs = portalConfig.portalSupportedEnvs();
+    for (Env env : portalEnvs) {
+      // 根据环境初始化查看角色
+      createAppViewEnvRole(appId, operator, env.toString());
+    }
+
+
     //assign master role to user
-    rolePermissionService
-        .assignRoleToUsers(RoleUtils.buildAppMasterRoleName(appId), Sets.newHashSet(app.getOwnerName()),
-            operator);
+    rolePermissionService.assignRoleToUsers(RoleUtils.buildAppMasterRoleName(appId), Sets.newHashSet(app.getOwnerName()), operator);
+    // 授予查看角色
+    rolePermissionService.assignRoleToUsers(RoleUtils.buildViewAppRoleName(appId), Sets.newHashSet(app.getOwnerName()), operator);
 
     initNamespaceRoles(appId, ConfigConsts.NAMESPACE_APPLICATION, operator);
     initNamespaceEnvRoles(appId, ConfigConsts.NAMESPACE_APPLICATION, operator);
 
     //assign modify、release namespace role to user
-    rolePermissionService.assignRoleToUsers(
-        RoleUtils.buildNamespaceRoleName(appId, ConfigConsts.NAMESPACE_APPLICATION, RoleType.MODIFY_NAMESPACE),
-        Sets.newHashSet(operator), operator);
-    rolePermissionService.assignRoleToUsers(
-        RoleUtils.buildNamespaceRoleName(appId, ConfigConsts.NAMESPACE_APPLICATION, RoleType.RELEASE_NAMESPACE),
-        Sets.newHashSet(operator), operator);
+    rolePermissionService.assignRoleToUsers(RoleUtils.buildNamespaceRoleName(appId, ConfigConsts.NAMESPACE_APPLICATION, RoleType.MODIFY_NAMESPACE), Sets.newHashSet(operator), operator);
```

Comment:
looks like you haven't imported the apollo code style - https://github.com/ctripcorp/apollo/tree/master/apollo-buildtools/style